spec/models/database_spec.rb

```diff
@@ -0,0 +1,145 @@
+    model_classes.reject!(&filter)
+
+    ActiveRecord::Base.descendants.each do |model_class|
+      next unless model_classes.include?(model_class.name)
+
+      model_class.reflect_on_all_associations(:belongs_to).each do |association|
+        migration = process_association(model_class, association, previous_models)
+        migrations << migration unless migration.nil?
+      end
+    end
+
+    print_missing_foreign_key_warnings(migrations)
+
+    puts "The following models are marked as todo in #{__FILE__}:"
+    puts pending_models.join(", ")
+
+    migrations
+  end
+
+  def print_missing_foreign_key_warnings(migrations)
+    return if migrations.empty?
+
+    puts "Foreign key(s) appear to be absent from the database. " \
+         "You can add it/them using the following migration(s):"
+    puts migrations.join("\n")
+    puts "\nTo disable this warning, add the class name(s) of the model(s) to models_todo " \
+         "in #{__FILE__}"
+
+    return if ENV.fetch("OFN_WRITE_FOREIGN_KEY_MIGRATIONS", false)
+
+    puts "Migrations have not been written to disk. To write migrations to disk, please " \
+         "add OFN_WRITE_FOREIGN_KEY_MIGRATIONS=true to the file .env.test.local"
+  end
+
+  def process_association(model_class, association, previous_models)
+    return if association.options[:polymorphic]
+
+    foreign_key_table_name = determine_foreign_key_table_name(model_class, association)
+    foreign_key_column = "#{association.options[:foreign_key] || association.name}_id"
+
+    # Filter out duplicate migrations
+    return if duplicate_migration?(model_class, foreign_key_table_name, previous_models)
+
+    previous_models[model_class.table_name] ||= []
+    previous_models[model_class.table_name] << foreign_key_table_name
+
+    generate_migration(model_class, association, foreign_key_table_name, foreign_key_column)
+  end
+
+  def determine_foreign_key_table_name(model_class, association)
+    if association.options[:class_name]
+      class_name = association.options[:class_name].underscore.parameterize
+      foreign_key_table_name = class_name.tableize
+    else
+      foreign_key_table_name = association.class_name.underscore.parameterize.tableize
+      namespace = model_class.name.deconstantize
+
+      unless association.class_name.deconstantize == namespace || namespace == "" ||
+             ActiveRecord::Base.connection.table_exists?(foreign_key_table_name)
+        foreign_key_table_name = "#{namespace.underscore}_#{foreign_key_table_name}"
+      end
+    end
+
+    foreign_key_table_name
+  end
+
+  def generate_migration(model_class, _association, foreign_key_table_name, foreign_key_column)
+    migration_name = "add_foreign_key_to_#{model_class.table_name}_#{foreign_key_table_name}"
+    migration_class_name = migration_name.camelize
+    migration_file_name = "db/migrate/#{Time.now.utc.strftime('%Y%m%d%H%M%S%L')}_" \
+                          "#{migration_name}.rb"
```

Comment:
I think it has to follow a timestamp so that the migrations are applied in order, [the timestamp is how Rails identifies the migrations internally](https://guides.rubyonrails.org/v3.2/migrations.html#:~:text=1.2%20What%E2%80%99s%20in%20a%20Name). Maybe I can keep track of the last version number and increment it if they are identical. That will guarantee uniqueness and preserve versioning.

My hope is that we will only need to generate one maybe two of these at a time from now on.